components/results_table/commons/results_table.lua

```diff
@@ -101,8 +108,201 @@ function BaseResultsTable:getQueryType()
 	error('Invalid querytype "' .. (args.querytype or '') .. '"')
 end
 
+function BaseResultsTable:create()
+	local data = self.args.data or self:queryData()
+
+	if Table.isEmpty(data) then
+		return
+	end
+
+	table.sort(data, function(placement1, placement2) return placement1.date > placement2.date end)
+
+	if self.config.onlyAchievements then
+		self.data = {data}
+		return self
+	end
+
+	-- split placements into years for non achievements
+	local splitData = Array.groupBy(data, function(placementData)
+		return placementData.date:sub(1,4)
+	end)
+
+	-- Set the header
+	Table.iter.forEach(splitData, function(dataSet)
+		dataSet.header = dataSet[1].date:sub(1,4)
+	end)
```

Comment:
```suggestion
	-- split placements into years for non achievements
	local splitData = Table.groupBy(data, function(_, placementData)
		return placementData.date:sub(1,4)
	end)

	-- Set the header
	Table.iter.forEachPair(splitData, function(year, dataSet)
		dataSet.header = year
	end)
```
Should work

Reply:
then i would have to make it an array afterwards

Reply:
nope doesn't work
it pushes it with the orig keys into the grouped subtables
and hence i can not run through it with ipairs later on

![Screenshot 2022-11-09 16 36 59](https://user-images.githubusercontent.com/75081997/200873894-3ae11623-ab2d-48b3-8c8f-f19f663abb96.png)

![Screenshot 2022-11-09 16 36 28](https://user-images.githubusercontent.com/75081997/200873978-f431dfa7-e0b2-4473-aae9-8e588259e766.png)